### ISSUE TITLE
Quests replaced by "New Quest"

### ISSUE THREAD

**Issue by j2ghz** [GTNewHorizons/NewHorizons#909]
Some of my quests disappeared and were replaced by `New Quest`. It also unlocked tiers I should have no access to. The quests are empty with no description, requirement or reward.
![](http://i.imgur.com/x1Ufs1s.png)
![](http://i.imgur.com/E9Fnkhm.png)

**Comment by j2ghz** [comment]
Feel free to close this, I guess it was caused by generating the world with BIOMESOP, then deleting chunks, manually editing NBT data, switching the save between server and singleplayer. Now i'm getting cast exceptions for the worldgen so I'm discarting this save.

**Comment by Dream-Master** [comment]
Ok

**Comment by j2ghz** [comment]
Happened again, possible causes: force closing minecraft, moving a save between singleplayer and server.
It seems to corrupt `QuestDatabase.json`
Example (normal save):
```
    {
      "questID": 239,
      "name": "Yet another brick in the wall...",
      "description": "A magic one this time. You are on your way to infusion, and you need 8 of those in total.",
      "isMain": true,
      "isSilent": false,
      "lockedProgress": false,
      "simultaneous": false,
      "globalParticipation": 0.0,
      "globalQuest": false,
      "globalShare": true,
      "autoClaim": false,
      "repeatTime": -1,
      "sounds": {
        "complete": "random.levelup",
        "update": "random.levelup"
      },
      "logic": "AND",
      "taskLogic": "AND",
      "icon": {
        "id": "Thaumcraft:blockCosmeticSolid",
        "Count": 1,
        "OreDict": "",
        "Damage": 7
      },
      "visibility": "UNLOCKED",
      "tasks": [
        {
          "partialMatch": true,
          "ignoreNBT": false,
          "consume": false,
          "autoConsume": false,
          "requiredItems": [
            {
              "id": "Thaumcraft:blockCosmeticSolid",
              "Count": 8,
              "OreDict": "",
              "Damage": 7
            }
          ],
          "taskID": "bq_standard:retrieval"
        }
      ],
      "rewards": [
        {
          "rewards": [
            {
              "id": "enhancedlootbags:lootbag",
              "Count": 1,
              "OreDict": "",
              "Damage": 9
            }
          ],
          "rewardID": "bq_standard:item"
        },
        {
          "rewards": [
            {
              "id": "dreamcraft:item.CoinDarkWizard",
              "Count": 10,
              "OreDict": "",
              "Damage": 0
            }
          ],
          "rewardID": "bq_standard:item"
        }
      ],
      "preRequisites": [
        251
      ]
    },
```
Example (corrupt save):
```
    {
      "questID": 239,
      "name": "New Quest",
      "description": "No Description",
      "isMain": false,
      "isSilent": false,
      "lockedProgress": false,
      "simultaneous": false,
      "globalParticipation": 0.0,
      "globalQuest": false,
      "globalShare": true,
      "autoClaim": false,
      "repeatTime": -1,
      "sounds": {
        "complete": "random.levelup",
        "update": "random.levelup"
      },
      "logic": "AND",
      "taskLogic": "AND",
      "icon": {
        "id": "minecraft:nether_star",
        "Count": 1,
        "OreDict": "",
        "Damage": 0
      },
      "visibility": "NORMAL",
      "tasks": [],
      "rewards": [],
      "preRequisites": []
    },
```

**Comment by Dream-Master** [comment]
QuestDatabase.json is in the cfg folder. Each World have own files in the save world folder like Quests, progression and party. Maybe something went wrong with sync something. If you using a never version of the Quest file just put it on the server in the cfg folder and load it with this command /bg_admin default load. This will put the new Quests into the files located in the Save Folder. Dit you run GTNH 1.2.1.0?

**Comment by j2ghz** [comment]
I've always been running 1.2.1.0. I replaced the file by the default one and it works, except some quests are completed even though they shouldn't be (the same ones that were replaced by new quest before) I guess they had no prerequisites so they were saved as completed. So I'll just remove those in progression and everything will be fixed. I think it's caused by minecraft crashing while saving (because chrome opened and I  ran out of RAM). I guess there's nothing for you to fix, I leave this here for others as a guide if they encounter the same issue.
